community/cypher/cypher-compiler-2.3/src/main/scala/org/neo4j/cypher/internal/compiler/v2_3/executionplan/CompiledExecutionResult.scala

```diff
@@ -138,7 +138,7 @@ abstract class CompiledExecutionResult(taskCloser: TaskCloser, statement:Stateme
 
   override def notifications = Iterable.empty[InternalNotification]
 
-  override def hasNext = innerIterator.hasNext
+  override def hasNext = !taskCloser.isClosed && innerIterator.hasNext
```

Comment:
This is not correct. The idea is to load the `innerIterator` on the first usage and then we expect statement to be closed. So basically `taskCloser.isClosed` should almost always be `true` here.